<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.269">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Website</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Website</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link active" href="./index.html" aria-current="page">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./about.html">
 <span class="menu-text">About</span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#gtex-pipeline-for-aaa-rnaseq-analysis" id="toc-gtex-pipeline-for-aaa-rnaseq-analysis" class="nav-link active" data-scroll-target="#gtex-pipeline-for-aaa-rnaseq-analysis">GTEx Pipeline for AAA RNAseq Analysis</a>
  <ul class="collapse">
  <li><a href="#install-rnaseqc-and-samtools-then-modify-.bashrc-file-to-make-them-callable-from-anywhere" id="toc-install-rnaseqc-and-samtools-then-modify-.bashrc-file-to-make-them-callable-from-anywhere" class="nav-link" data-scroll-target="#install-rnaseqc-and-samtools-then-modify-.bashrc-file-to-make-them-callable-from-anywhere">Install RNAseQC and samtools, then modify .bashrc file to make them callable from anywhere:</a></li>
  <li><a href="#add-to-homegerard.bashrc-file" id="toc-add-to-homegerard.bashrc-file" class="nav-link" data-scroll-target="#add-to-homegerard.bashrc-file">Add to /home/gerard/.bashrc file:</a></li>
  <li><a href="#in-r-create-a-list-with-the-files-that-will-be-used-in-the-pipeline-steps" id="toc-in-r-create-a-list-with-the-files-that-will-be-used-in-the-pipeline-steps" class="nav-link" data-scroll-target="#in-r-create-a-list-with-the-files-that-will-be-used-in-the-pipeline-steps">In R, create a list with the files that will be used in the pipeline steps:</a></li>
  <li><a href="#prepare-array-for-the-star-step-where-we-need-the-name-to-include-the-_s" id="toc-prepare-array-for-the-star-step-where-we-need-the-name-to-include-the-_s" class="nav-link" data-scroll-target="#prepare-array-for-the-star-step-where-we-need-the-name-to-include-the-_s">Prepare array for the STAR step, where we need the name to include the “_S#“:</a></li>
  <li><a href="#now-prepare-array-for-the-other-steps-without-the-_s" id="toc-now-prepare-array-for-the-other-steps-without-the-_s" class="nav-link" data-scroll-target="#now-prepare-array-for-the-other-steps-without-the-_s">Now, prepare array for the other steps, without the “_S#“:</a></li>
  <li><a href="#to-set-the-array_star.txt-file-as-array-in-bash" id="toc-to-set-the-array_star.txt-file-as-array-in-bash" class="nav-link" data-scroll-target="#to-set-the-array_star.txt-file-as-array-in-bash">To set the Array_STAR.txt file as array in bash:</a></li>
  <li><a href="#bam-to-fastq-conversion.-we-receive-fastq-files-from-the-sequencing-company-therefore-the-first-step-in-the-gtex-pipeline-is-not-required." id="toc-bam-to-fastq-conversion.-we-receive-fastq-files-from-the-sequencing-company-therefore-the-first-step-in-the-gtex-pipeline-is-not-required." class="nav-link" data-scroll-target="#bam-to-fastq-conversion.-we-receive-fastq-files-from-the-sequencing-company-therefore-the-first-step-in-the-gtex-pipeline-is-not-required.">BAM to FASTQ conversion. We receive FASTQ files from the sequencing company, therefore, the first step in the GTEx pipeline is not required.</a></li>
  <li><a href="#build-the-star-index-using-the-gencode-version-v26-and-for-reads-of-length-149.-only-one-time-to-create-the-star-index" id="toc-build-the-star-index-using-the-gencode-version-v26-and-for-reads-of-length-149.-only-one-time-to-create-the-star-index" class="nav-link" data-scroll-target="#build-the-star-index-using-the-gencode-version-v26-and-for-reads-of-length-149.-only-one-time-to-create-the-star-index">Build the STAR index, using the gencode version v26, and for reads of length 149. Only one time, to create the STAR index:</a></li>
  <li><a href="#build-the-rsem-index.-only-one-time-to-create-the-rsem-index" id="toc-build-the-rsem-index.-only-one-time-to-create-the-rsem-index" class="nav-link" data-scroll-target="#build-the-rsem-index.-only-one-time-to-create-the-rsem-index">Build the RSEM index. Only one time, to create the RSEM index:</a></li>
  <li><a href="#star-alignment" id="toc-star-alignment" class="nav-link" data-scroll-target="#star-alignment">STAR alignment:</a></li>
  <li><a href="#to-set-the-array_others.txt-file-as-array-in-bash" id="toc-to-set-the-array_others.txt-file-as-array-in-bash" class="nav-link" data-scroll-target="#to-set-the-array_others.txt-file-as-array-in-bash">To set the Array_Others.txt file as array in bash:</a></li>
  <li><a href="#after-obtaining-the-sam-aligned-files-we-run-rseqc-with-one-of-them-to-identify-the-type-of-data-we-have" id="toc-after-obtaining-the-sam-aligned-files-we-run-rseqc-with-one-of-them-to-identify-the-type-of-data-we-have" class="nav-link" data-scroll-target="#after-obtaining-the-sam-aligned-files-we-run-rseqc-with-one-of-them-to-identify-the-type-of-data-we-have">After obtaining the SAM aligned files, we run RSeQC with one of them to identify the type of data we have:</a></li>
  <li><a href="#indicates-that-is-paired-end-stranded." id="toc-indicates-that-is-paired-end-stranded." class="nav-link" data-scroll-target="#indicates-that-is-paired-end-stranded.">Indicates that is paired-end STRANDED.</a></li>
  <li><a href="#sync-bams-optional-copy-qc-flags-and-read-group-ids-not-done" id="toc-sync-bams-optional-copy-qc-flags-and-read-group-ids-not-done" class="nav-link" data-scroll-target="#sync-bams-optional-copy-qc-flags-and-read-group-ids-not-done">Sync BAMs (optional; copy QC flags and read group IDs): NOT DONE</a></li>
  <li><a href="#mark-duplicates-picard" id="toc-mark-duplicates-picard" class="nav-link" data-scroll-target="#mark-duplicates-picard">Mark duplicates (Picard):</a></li>
  <li><a href="#rna-seqc-i-am-not-using-the-python-script-descrbied-in-the-gtex-pipeline-because-it-is-not-detecting-the-.gtf-file-correctly.-i-am-using-the-software-directly-with-the-same-options-as-in-the-gtex-pipeline.-the-options-that-increase-the-number-of-exonic-reads-are-rf-and-using-gencode.v26.grch38.ercc.genes.collapsed_only.gtf-as-gencode-reference.-i-start-creating-the-rf-folder-in-each-of-the-samples-folders" id="toc-rna-seqc-i-am-not-using-the-python-script-descrbied-in-the-gtex-pipeline-because-it-is-not-detecting-the-.gtf-file-correctly.-i-am-using-the-software-directly-with-the-same-options-as-in-the-gtex-pipeline.-the-options-that-increase-the-number-of-exonic-reads-are-rf-and-using-gencode.v26.grch38.ercc.genes.collapsed_only.gtf-as-gencode-reference.-i-start-creating-the-rf-folder-in-each-of-the-samples-folders" class="nav-link" data-scroll-target="#rna-seqc-i-am-not-using-the-python-script-descrbied-in-the-gtex-pipeline-because-it-is-not-detecting-the-.gtf-file-correctly.-i-am-using-the-software-directly-with-the-same-options-as-in-the-gtex-pipeline.-the-options-that-increase-the-number-of-exonic-reads-are-rf-and-using-gencode.v26.grch38.ercc.genes.collapsed_only.gtf-as-gencode-reference.-i-start-creating-the-rf-folder-in-each-of-the-samples-folders">RNA-SeQC: I am not using the python script descrbied in the GTEx pipeline because it is not detecting the .gtf file correctly. I am using the software directly, with the same options as in the GTEx pipeline. The options that increase the number of exonic reads are: RF and using gencode.v26.GRCh38.ERCC.genes.collapsed_only.gtf as gencode reference. I start creating the RF folder in each of the samples folders</a></li>
  <li><a href="#rsem-transcript-quantification" id="toc-rsem-transcript-quantification" class="nav-link" data-scroll-target="#rsem-transcript-quantification">RSEM transcript quantification:</a></li>
  <li><a href="#sample-level-outputs-in-gct-format-can-be-concatenated-using-combine_gcts.py" id="toc-sample-level-outputs-in-gct-format-can-be-concatenated-using-combine_gcts.py" class="nav-link" data-scroll-target="#sample-level-outputs-in-gct-format-can-be-concatenated-using-combine_gcts.py">Sample-level outputs in GCT format can be concatenated using combine_GCTs.py:</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Website</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p>This is a Quarto website.</p>
<p>To learn more about Quarto websites visit <a href="https://quarto.org/docs/websites" class="uri">https://quarto.org/docs/websites</a>.</p>
<section id="gtex-pipeline-for-aaa-rnaseq-analysis" class="level2">
<h2 class="anchored" data-anchor-id="gtex-pipeline-for-aaa-rnaseq-analysis">GTEx Pipeline for AAA RNAseq Analysis</h2>
<section id="install-rnaseqc-and-samtools-then-modify-.bashrc-file-to-make-them-callable-from-anywhere" class="level3">
<h3 class="anchored" data-anchor-id="install-rnaseqc-and-samtools-then-modify-.bashrc-file-to-make-them-callable-from-anywhere">Install RNAseQC and samtools, then modify .bashrc file to make them callable from anywhere:</h3>
<p>mkdir opt cd opt mkdir rnaseqc cd rnadeqc wget https://github.com/getzlab/rnaseqc/releases/download/v2.3.6/rnaseqc.v2.3.6.linux.gz gunzip rnaseqc.v2.3.6.linux.gz &amp;&amp; mv rnaseqc.v2.3.6.linux rnaseqc &amp;&amp; chmod 775 rnaseqc PATH=$PATH:/home/gerard/opt/rnaseqc pip3 install rnaseqc</p>
</section>
<section id="add-to-homegerard.bashrc-file" class="level3">
<h3 class="anchored" data-anchor-id="add-to-homegerard.bashrc-file">Add to /home/gerard/.bashrc file:</h3>
<p>export PATH=<span class="math inline">\(PATH:/home/gerard/samtools-1.16.1 export PATH=\)</span>PATH:/home/gerard/opt/rnaseqc</p>
</section>
<section id="in-r-create-a-list-with-the-files-that-will-be-used-in-the-pipeline-steps" class="level3">
<h3 class="anchored" data-anchor-id="in-r-create-a-list-with-the-files-that-will-be-used-in-the-pipeline-steps">In R, create a list with the files that will be used in the pipeline steps:</h3>
<p>ff &lt;- list.files(path = “/home/datasets/aaa/AAA.d.RNAseq/FastQ”,pattern = “<em>R1_001.fastq.gz”, recursive = ”TRUE”, full.names = F) ff &lt;- gsub(”.</em>/”,““,ff) ff &lt;- sub(”^([^_]*_[^_]<em>).</em>”, “\1”, ff)</p>
</section>
<section id="prepare-array-for-the-star-step-where-we-need-the-name-to-include-the-_s" class="level3">
<h3 class="anchored" data-anchor-id="prepare-array-for-the-star-step-where-we-need-the-name-to-include-the-_s">Prepare array for the STAR step, where we need the name to include the “_S#“:</h3>
<p>write.table(ff, “/home/gerard/Array_STAR.txt”, quote = F, sep = ““, row.names = F, col.names = F)</p>
</section>
<section id="now-prepare-array-for-the-other-steps-without-the-_s" class="level3">
<h3 class="anchored" data-anchor-id="now-prepare-array-for-the-other-steps-without-the-_s">Now, prepare array for the other steps, without the “_S#“:</h3>
<p>ff &lt;- gsub(“_.*“,”“,ff)</p>
<p>#write.table(ff, “/home/gerard/Array_Others.txt”, quote = F, sep = ““, row.names = F, col.names = F)</p>
</section>
<section id="to-set-the-array_star.txt-file-as-array-in-bash" class="level3">
<h3 class="anchored" data-anchor-id="to-set-the-array_star.txt-file-as-array-in-bash">To set the Array_STAR.txt file as array in bash:</h3>
<p>mapfile -t arr &lt; /home/gerard/Array_STAR.txt</p>
</section>
<section id="bam-to-fastq-conversion.-we-receive-fastq-files-from-the-sequencing-company-therefore-the-first-step-in-the-gtex-pipeline-is-not-required." class="level3">
<h3 class="anchored" data-anchor-id="bam-to-fastq-conversion.-we-receive-fastq-files-from-the-sequencing-company-therefore-the-first-step-in-the-gtex-pipeline-is-not-required.">BAM to FASTQ conversion. We receive FASTQ files from the sequencing company, therefore, the first step in the GTEx pipeline is not required.</h3>
<p>for i in *.bam;do sample_id=${i:0:7} docker run –rm -v <span class="math inline">\(path_to_data:/data -t broadinstitute/gtex_rnaseq \
/bin/bash -c "/src/run_SamToFastq.py /data/\)</span>i -p ${sample_id} -o /data” done</p>
</section>
<section id="build-the-star-index-using-the-gencode-version-v26-and-for-reads-of-length-149.-only-one-time-to-create-the-star-index" class="level3">
<h3 class="anchored" data-anchor-id="build-the-star-index-using-the-gencode-version-v26-and-for-reads-of-length-149.-only-one-time-to-create-the-star-index">Build the STAR index, using the gencode version v26, and for reads of length 149. Only one time, to create the STAR index:</h3>
<p>mkdir STARv275a_genome_GRCh38_noALT_noHLA_noDecoy_v26_oh149</p>
<p>docker run –rm -v $path_to_data:/data -t broadinstitute/gtex_rnaseq:V10<br>
/bin/bash -c “STAR<br>
–runMode genomeGenerate<br>
–genomeDir /data/STARv275a_genome_GRCh38_noALT_noHLA_noDecoy_v26_oh149<br>
–genomeFastaFiles /data/Homo_sapiens_assembly38_noALT_noHLA_noDecoy_ERCC.fasta<br>
–sjdbGTFfile /data/gencode.v26.GRCh38.annotation.ERCC92.gtf<br>
–sjdbOverhang 149<br>
–runThreadN 16”</p>
</section>
<section id="build-the-rsem-index.-only-one-time-to-create-the-rsem-index" class="level3">
<h3 class="anchored" data-anchor-id="build-the-rsem-index.-only-one-time-to-create-the-rsem-index">Build the RSEM index. Only one time, to create the RSEM index:</h3>
<p>docker run –rm -v $path_to_data:/data -t broadinstitute/gtex_rnaseq:V10<br>
/bin/bash -c “rsem-prepare-reference<br>
/data/GTEx/Homo_sapiens_assembly38_noALT_noHLA_noDecoy_ERCC.fasta<br>
/data/GTEx/rsem_reference/rsem_reference<br>
–gtf /data/GTEx/gencode.v26.GRCh38.annotation.ERCC92.gtf<br>
–num-threads 16”</p>
</section>
<section id="star-alignment" class="level3">
<h3 class="anchored" data-anchor-id="star-alignment">STAR alignment:</h3>
<p>path_to_data=“/home/gerard/RNAseq”</p>
<p>for i in “<span class="math inline">\({arr[0]}";do folder=\)</span>{i:0:9} cp -r /home/datasets/aaa/AAA.d.RNAseq/FastQ/${folder} /home/gerard/RNAseq echo Folder ${folder} has been copied! docker run –rm -v <span class="math inline">\(path_to_data:/data -t broadinstitute/gtex_rnaseq:V10 \
/bin/bash -c "/src/run_STAR.py \
/data/GTEx/STARv275a_genome_GRCh38_noALT_noHLA_noDecoy_v26_oh149 \
/data/\)</span>{folder}/<span class="math inline">\({i}_R1_001.fastq.gz \
/data/\)</span>{folder}/${i}_R2_001.fastq.gz<br>
<span class="math inline">\(folder \
--threads 28 \
--output_dir /tmp/star_out &amp;&amp; mv /tmp/star_out /data/GTEx/\)</span>{folder}” rm -r /home/gerard/RNAseq/${folder} echo Folder ${folder} has been removed! done</p>
</section>
<section id="to-set-the-array_others.txt-file-as-array-in-bash" class="level3">
<h3 class="anchored" data-anchor-id="to-set-the-array_others.txt-file-as-array-in-bash">To set the Array_Others.txt file as array in bash:</h3>
<p>mapfile -t arr &lt; /home/gerard/Array_Others.txt</p>
</section>
<section id="after-obtaining-the-sam-aligned-files-we-run-rseqc-with-one-of-them-to-identify-the-type-of-data-we-have" class="level3">
<h3 class="anchored" data-anchor-id="after-obtaining-the-sam-aligned-files-we-run-rseqc-with-one-of-them-to-identify-the-type-of-data-we-have">After obtaining the SAM aligned files, we run RSeQC with one of them to identify the type of data we have:</h3>
<p>python3 ./anaconda3/bin/infer_experiment.py -r /home/gerard/RNAseq/GTEx/BED/hg38_RefSeq.bed -i /home/gerard/RNAseq/GTEx/22ID00809/22ID00809.Aligned.sortedByCoord.out.bam</p>
<p>Output:</p>
<p>Reading reference gene model /home/gerard/RNAseq/GTEx/BED/hg38_RefSeq.bed … Done Loading SAM/BAM file … Total 200000 usable reads were sampled</p>
<p>This is PairEnd Data Fraction of reads failed to determine: 0.0185 Fraction of reads explained by “1++,1–,2+-,2-+”: 0.0187 Fraction of reads explained by “1+-,1-+,2++,2–”: 0.9628</p>
</section>
<section id="indicates-that-is-paired-end-stranded." class="level3">
<h3 class="anchored" data-anchor-id="indicates-that-is-paired-end-stranded.">Indicates that is paired-end STRANDED.</h3>
</section>
<section id="sync-bams-optional-copy-qc-flags-and-read-group-ids-not-done" class="level3">
<h3 class="anchored" data-anchor-id="sync-bams-optional-copy-qc-flags-and-read-group-ids-not-done">Sync BAMs (optional; copy QC flags and read group IDs): NOT DONE</h3>
<p>for i in “${arr[@]}”;do docker run –rm -v <span class="math inline">\(path_to_data:/data -t broadinstitute/gtex_rnaseq:V10 \
/bin/bash -c "/src/run_bamsync.sh \
/data/\)</span>input_bam<br>
/data/<span class="math inline">\({sample_id}/\)</span>{sample_id}.Aligned.sortedByCoord.out.bam<br>
/data/<span class="math inline">\({sample_id}/\)</span>{sample_id}” done</p>
</section>
<section id="mark-duplicates-picard" class="level3">
<h3 class="anchored" data-anchor-id="mark-duplicates-picard">Mark duplicates (Picard):</h3>
<p>path_to_data=“/home/gerard/RNAseq/GTEx”</p>
<p>for i in “${arr[@]}”;do docker run –rm -v <span class="math inline">\(path_to_data:/data -t broadinstitute/gtex_rnaseq:V10 \
/bin/bash -c "/src/run_MarkDuplicates.py \
/data/\)</span>{i}/${i}.Aligned.sortedByCoord.out.bam<br>
<span class="math inline">\({i}.Aligned.sortedByCoord.out.patched.md \
--output_dir /data/\)</span>{i}” done</p>
</section>
<section id="rna-seqc-i-am-not-using-the-python-script-descrbied-in-the-gtex-pipeline-because-it-is-not-detecting-the-.gtf-file-correctly.-i-am-using-the-software-directly-with-the-same-options-as-in-the-gtex-pipeline.-the-options-that-increase-the-number-of-exonic-reads-are-rf-and-using-gencode.v26.grch38.ercc.genes.collapsed_only.gtf-as-gencode-reference.-i-start-creating-the-rf-folder-in-each-of-the-samples-folders" class="level3">
<h3 class="anchored" data-anchor-id="rna-seqc-i-am-not-using-the-python-script-descrbied-in-the-gtex-pipeline-because-it-is-not-detecting-the-.gtf-file-correctly.-i-am-using-the-software-directly-with-the-same-options-as-in-the-gtex-pipeline.-the-options-that-increase-the-number-of-exonic-reads-are-rf-and-using-gencode.v26.grch38.ercc.genes.collapsed_only.gtf-as-gencode-reference.-i-start-creating-the-rf-folder-in-each-of-the-samples-folders">RNA-SeQC: I am not using the python script descrbied in the GTEx pipeline because it is not detecting the .gtf file correctly. I am using the software directly, with the same options as in the GTEx pipeline. The options that increase the number of exonic reads are: RF and using gencode.v26.GRCh38.ERCC.genes.collapsed_only.gtf as gencode reference. I start creating the RF folder in each of the samples folders</h3>
<p>for i in “<span class="math inline">\({arr[@]}";do mkdir /home/gerard/RNAseq/GTEx/\)</span>{i}/RF rnaseqc gencode.v26.GRCh38.ERCC.genes.collapsed_only.gtf /home/gerard/RNAseq/GTEx/<span class="math inline">\({i}/\)</span>{i}.Aligned.sortedByCoord.out.md.bam /home/gerard/RNAseq/GTEx/${i}/RF<br>
-s ${i} –stranded RF -vv done</p>
</section>
<section id="rsem-transcript-quantification" class="level3">
<h3 class="anchored" data-anchor-id="rsem-transcript-quantification">RSEM transcript quantification:</h3>
<p>path_to_data=“/home/gerard/RNAseq”</p>
<p>for i in “${arr[@]}”;do docker run –rm -v <span class="math inline">\(path_to_data:/data -t broadinstitute/gtex_rnaseq:V10 \
/bin/bash -c "/src/run_RSEM.py \
/data/GTEx/rsem_reference \
/data/GTEx/\)</span>{i}/<span class="math inline">\({i}.Aligned.toTranscriptome.out.bam \
/data/GTEx/\)</span>{i}/<span class="math inline">\({i} \
--o /data/GTEx/\)</span>{i}<br>
–is_stranded true<br>
–threads 24” done</p>
</section>
<section id="sample-level-outputs-in-gct-format-can-be-concatenated-using-combine_gcts.py" class="level3">
<h3 class="anchored" data-anchor-id="sample-level-outputs-in-gct-format-can-be-concatenated-using-combine_gcts.py">Sample-level outputs in GCT format can be concatenated using combine_GCTs.py:</h3>
<p>docker run –rm -v $path_to_data:/data -t broadinstitute/gtex_rnaseq:V10<br>
/bin/bash -c “python3 /src/combine_GCTs.py<br>
${rnaseqc_tpm_gcts} ${sample_set_id}.rnaseqc_tpm”</p>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>